src/exercises/FeedbackButtons.js

```diff
@@ -1,38 +1,136 @@
 import strings from "../i18n/definitions";
 
-const buttons = [
-  { name: "Too Easy", value: "too_easy" },
-  { name: "Too Hard", value: "too_hard" },
-  { name: "Bad example", value: "not_a_good_example" },
-  { name: "Bad word", value: "bad_word" },
-  { name: "Other", value: "dont_show_it_to_me_again" },
-];
-export default function FeedbackButtons({ show, setShow, feedbackFunction }) {
-  function toggleShow() {
-    setShow(!show);
+export default function FeedbackButtons({
+  feedbackFunction,
+  currentExerciseType,
+  currentBookmarkToStudy,
+}) {
+  const buttons = [
+    { name: "Too Easy", value: "too_easy" },
+    { name: "Too Hard", value: "too_hard" },
+    { name: "Bad Translation", value: "bad_translation" },
+    { name: "Other", value: "other" },
+  ];
+
+  if (currentExerciseType !== "Match_three_L1W_to_three_L2W") {
+    buttons.splice(3, 0, { name: "Bad Context", value: "not_a_good_context" });
+  }
+  const [showInput, setShowInput] = useState(false);
+  const [className, setClassName] = useState("");
+  const [input, setInput] = useState("");
+  const [selectedId, setSelectedId] = useState();
+
+  useEffect(() => {
+    if (currentExerciseType !== "Match_three_L1W_to_three_L2W") {
+      setSelectedId(currentBookmarkToStudy.id);
+    }
+    console.log(selectedId);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  function isIterable(obj) {
+    if (obj == null) {
+      return false;
+    }
+    return typeof obj[Symbol.iterator] === "function";
+  }
+
+  function buttonClick(value) {
+    if (!selectedId) {
+      alert("Please select the word(s) for which you are providing feedback.");
+    } else {
+      if (value !== "other") {
+        feedbackFunction(value, selectedId);
+      } else {
+        setClassName("selected");
+        setShowInput(true);
+      }
+    }
+  }
+
+  function handleTyping(event) {
+    setInput(event.target.value);
   }
+
+  function handleSelection(event) {
+    setSelectedId(Number(event.target.value));
+  }
+
+  function handleSubmit(event) {
+    if (input === "") {
+      alert("Please type your feedback before submitting.");
+      event.preventDefault();
+    } else {
+      let re1 = /[.,'´`?!:;]/g;
+      let re2 = /\s{2,}/g;
+      const newFeedback = input
+        .toLowerCase()
+        .trim()
+        .replace(re1, "")
+        .replace(re2, "")
+        .replaceAll(" ", "_");
+      feedbackFunction(newFeedback, selectedId);
+      setInput("");
+      event.preventDefault();
+    }
+  }
+
   return (
-    <div>
-      <s.FeedbackLinkHolder>
-        <Link to={"#"} className="discrete-link" onClick={toggleShow}>
-          {strings.giveFeedback}
-        </Link>
-      </s.FeedbackLinkHolder>
-
-      {show && (
-        <s.FeedbackButtonsHolder>
-          {buttons.map((each) => (
-            <button
+    <s.FeedbackHolder>
+      {currentExerciseType === "Match_three_L1W_to_three_L2W" &&
+        isIterable(currentBookmarkToStudy) && (
```

Comment:
looking at this, it feels like we're doing something wrong.
either the code is wrong, or the name of that prop is wrong.
how can a bookmark be iterable? 

could it be that the `currentBookmarkToStudy` should be `currentBookmarks`? Then you need to simply test the length of the list: it can be either 1 or 3. And then you don't need the isIterable function anymore?

Reply:
Renamed `currentBookmarkToStudy` / `bookmarkToStudy` into `currentBookmarksToStudy` / `bookmarksToStudy`.